
Note:  MRC does not permit #include <xxx> -
the < > is not allowed - the two .rs here simply replace
the < > with " "
? capprsrc.r
? ccorersrc.r
Index: app.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/app.cpp,v
retrieving revision 1.132
diff -b -u -2 -r1.132 app.cpp
--- app.cpp	2003/12/28 22:46:08	1.132
+++ app.cpp	2004/01/29 19:51:34
@@ -43,5 +43,5 @@
 // mac
 
-#ifndef __DARWIN__
+#if !defined(__DARWIN__) && !defined(__MRC__) 
   #if __option(profile)
     #include <profiler.h>
@@ -589,5 +589,5 @@
   }
 
-#ifndef __DARWIN__
+#if !defined(__MRC__) && !defined(__DARWIN__)
 #  if __option(profile)
     ProfilerInit( collectDetailed, bestTimeBase , 40000 , 50 ) ;
@@ -699,5 +699,5 @@
     wxWinMacControlList = NULL;
 
-#ifndef __DARWIN__
+#if !defined(__DARWIN__) && !defined(__MRC__)
 #  if __option(profile)
     ProfilerDump( (StringPtr)"\papp.prof" ) ;
Index: bitmap.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/bitmap.cpp,v
retrieving revision 1.44
diff -b -u -2 -r1.44 bitmap.cpp
--- bitmap.cpp	2003/09/20 15:38:43	1.44
+++ bitmap.cpp	2004/01/29 19:51:34
@@ -22,6 +22,4 @@
 #include "wx/xpmdecod.h"
 
-#include "wx/rawbmp.h"
-
 IMPLEMENT_DYNAMIC_CLASS(wxBitmap, wxGDIObject)
 IMPLEMENT_DYNAMIC_CLASS(wxMask, wxObject)
@@ -1328,5 +1326,5 @@
     AddHandler(new wxICONResourceHandler) ;
 }
-
+#ifndef __MRC__
 // ----------------------------------------------------------------------------
 // raw bitmap access support
@@ -1423,5 +1421,5 @@
     }
 }
-
+#endif
 void wxBitmap::UseAlpha()
 {
Index: button.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/button.cpp,v
retrieving revision 1.28
diff -b -u -2 -r1.28 button.cpp
--- button.cpp	2003/11/18 17:52:25	1.28
+++ button.cpp	2004/01/29 19:51:34
@@ -131,2 +131,8 @@
 }
 
+/*static*/	//req'd on MPW
+wxSize wxButtonBase::GetDefaultSize()
+{
+	return wxDefaultSize;
+}
+
Index: combobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/combobox.cpp,v
retrieving revision 1.30
diff -b -u -2 -r1.30 combobox.cpp
--- combobox.cpp	2003/11/23 08:17:57	1.30
+++ combobox.cpp	2004/01/29 19:51:34
@@ -412,8 +412,8 @@
     return m_choice->DoInsert( item , pos ) ;
 }
-
+//MRC does not support return in void funcs
 void wxComboBox::DoSetItemClientData(int n, void* clientData) 
 {
-    return m_choice->DoSetItemClientData( n , clientData ) ;
+	m_choice->DoSetItemClientData( n , clientData ) ;
 }
 
@@ -425,5 +425,5 @@
 void wxComboBox::DoSetItemClientObject(int n, wxClientData* clientData)
 {
-    return m_choice->DoSetItemClientObject( n , clientData ) ;
+    m_choice->DoSetItemClientObject( n , clientData ) ;
 }
 
Index: cursor.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/cursor.cpp,v
retrieving revision 1.15
diff -b -u -2 -r1.15 cursor.cpp
--- cursor.cpp	2003/03/27 20:29:51	1.15
+++ cursor.cpp	2004/01/29 19:51:34
@@ -73,4 +73,5 @@
     }
 }
+#if wxUSE_IMAGE
 
 // Cursors
@@ -126,5 +127,4 @@
     return retval ;
 }
-
 void wxCursor::CreateFromImage(const wxImage & image) 
 {
@@ -497,4 +497,5 @@
 }
 
+#endif //wxUSE_IMAGE
 // Global cursor setting
 void wxSetCursor(const wxCursor& cursor)
Index: dc.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/dc.cpp,v
retrieving revision 1.85
diff -b -u -2 -r1.85 dc.cpp
--- dc.cpp	2003/12/11 09:05:45	1.85
+++ dc.cpp	2004/01/29 19:51:34
@@ -37,4 +37,7 @@
 #endif
 
+#ifdef __MRC__
+#include <math.h>
+#endif
 //-----------------------------------------------------------------------------
 // constants
@@ -1411,9 +1414,10 @@
         {
             sizeof( Fixed ) ,
-        } ;
-        ATSUAttributeValuePtr    atsuValues[sizeof(atsuTags)/sizeof(ATSUAttributeTag)] =
-        {
-            &atsuAngle ,
         } ;
+        ATSUAttributeValuePtr    atsuValues[sizeof(atsuTags)/sizeof(ATSUAttributeTag)];
+		//MRC only supports initializing arrays with constant values...
+		atsuValues[0] =
+            &atsuAngle;
+			
         status = ::ATSUSetLayoutControls(atsuLayout , sizeof(atsuTags)/sizeof(ATSUAttributeTag),
             atsuTags, atsuSizes, atsuValues ) ;
@@ -1843,17 +1847,15 @@
     //BslnBaselineClass kBaselineDefault = kBSLNHangingBaseline ;
     ATSUVerticalCharacterType kHorizontal = kATSUStronglyHorizontal;
-    ATSUAttributeValuePtr    atsuValues[sizeof(atsuTags)/sizeof(ATSUAttributeTag)] =
-    {
-        &atsuFont ,
-            &atsuSize ,
+    ATSUAttributeValuePtr    atsuValues[sizeof(atsuTags)/sizeof(ATSUAttributeTag)];
+	atsuValues[0]=&atsuFont;
+	atsuValues[1]=&atsuSize;
+    atsuValues[2]=&kHorizontal;
+	atsuValues[3]=(qdStyle & bold) ? &kTrue : &kFalse;
+	atsuValues[4]=(qdStyle & italic) ? &kTrue : &kFalse;
+	atsuValues[5]=(qdStyle & underline) ? &kTrue : &kFalse;
+	atsuValues[6]=(qdStyle & condense) ? &kTrue : &kFalse;
+	atsuValues[7]=(qdStyle & extend) ? &kTrue : &kFalse;        
             //        &MAC_WXCOLORREF( m_textForegroundColour.GetPixel() ) ,
             //        &kBaselineDefault ,
-            &kHorizontal,
-            (qdStyle & bold) ? &kTrue : &kFalse ,
-            (qdStyle & italic) ? &kTrue : &kFalse ,
-            (qdStyle & underline) ? &kTrue : &kFalse ,
-            (qdStyle & condense) ? &kTrue : &kFalse ,
-            (qdStyle & extend) ? &kTrue : &kFalse ,
-    } ;
     status = ::ATSUSetAttributes((ATSUStyle)m_macATSUIStyle, sizeof(atsuTags)/sizeof(ATSUAttributeTag) ,
         atsuTags, atsuSizes, atsuValues);
 
Index: filedlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/filedlg.cpp,v
retrieving revision 1.40
diff -b -u -2 -r1.40 filedlg.cpp
--- filedlg.cpp	2003/12/12 02:19:00	1.40
+++ filedlg.cpp	2004/01/29 19:51:34
@@ -217,10 +217,10 @@
 
         const size_t extCount = myData->extensions.GetCount();
-        for ( size_t i = 0 ; i < extCount; i++ )
+        for ( size_t i2 = 0 ; i2 < extCount; i2++ )
         {
             int j ;
             for ( j = 0 ; gfilters[j] ; j++ )
             {
-                if ( myData->extensions[i] == gfilters[j]  )
+                if ( myData->extensions[i2] == gfilters[j]  )
                 {
                     myData->filtermactypes.Add( gfiltersmac[j] ) ;
Index: gsocket.c
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/gsocket.c,v
retrieving revision 1.17
diff -b -u -2 -r1.17 gsocket.c
--- gsocket.c	2003/08/10 16:22:08	1.17
+++ gsocket.c	2004/01/29 19:51:54
@@ -31,5 +31,11 @@
   #endif
 #else
+#ifndef __MRC__
   #include <MacHeaders.c>
+#else /*__MRC__*/
+	#define true 1
+	#define false 0
+	#include "wx/mrc.h"
+#endif
   #define OTUNIXERRORS 1
   #include <OpenTransport.h>
@@ -38,5 +44,5 @@
 #endif
 #if TARGET_CARBON && !defined(OTAssert)
-  #define OTAssert( str , cond ) /* does not exists in Carbon */
+  #define OTAssert( str , cond ) /* does not exist in Carbon */
 #endif
 
@@ -44,10 +50,12 @@
 #include <errno.h>
 #include <string.h>
-#include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <stddef.h>
 #include <ctype.h>
-#include <utime.h>
+#ifndef __MRC__
+	#include <unistd.h>
+	#include <utime.h>
+#endif
 
 /*
Index: mdi.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/mdi.cpp,v
retrieving revision 1.20
diff -b -u -2 -r1.20 mdi.cpp
--- mdi.cpp	2003/12/23 12:39:33	1.20
+++ mdi.cpp	2004/01/29 19:51:34
@@ -288,5 +288,5 @@
 void wxMDIChildFrame::SetMenuBar(wxMenuBar *menu_bar)
 {
-    return wxFrame::SetMenuBar( menu_bar ) ;
+    wxFrame::SetMenuBar( menu_bar ) ;
 }
 
Index: msgdlg.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/msgdlg.cpp,v
retrieving revision 1.18
diff -b -u -2 -r1.18 msgdlg.cpp
--- msgdlg.cpp	2003/10/10 04:26:45	1.18
+++ msgdlg.cpp	2004/01/29 19:51:34
@@ -50,5 +50,5 @@
         alertType = kAlertCautionAlert ;
     
-#if TARGET_CARBON
+#if TARGET_CARBON && defined(__DARWIN__) //CreateStandardAlert only available on OSX+
     if ( UMAGetSystemVersion() >= 0x1000 )
     {
Index: notebmac.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/notebmac.cpp,v
retrieving revision 1.38
diff -b -u -2 -r1.38 notebmac.cpp
--- notebmac.cpp	2003/09/10 07:29:49	1.38
+++ notebmac.cpp	2004/01/29 19:51:34
@@ -454,5 +454,6 @@
             // of static counter
             const wxBitmap* bmap = GetImageList()->GetBitmap( GetPageImage(ii ) ) ;
-            if ( bmap )
+#if wxUSE_IMAGE
+            if ( bmap)
             {
                 wxBitmap scaledBitmap ;
@@ -479,4 +480,5 @@
                 DisposeHandle( (Handle) iconFamily ) ;
             }
+#endif //wxUSE_IMAGE
         }
 #endif
Index: pnghand.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/pnghand.cpp,v
retrieving revision 1.17
diff -b -u -2 -r1.17 pnghand.cpp
--- pnghand.cpp	2003/09/25 23:10:35	1.17
+++ pnghand.cpp	2004/01/29 19:51:34
@@ -9,5 +9,5 @@
 // Licence:     wxWindows licence
 /////////////////////////////////////////////////////////////////////////////
-
+#if 0
 #ifdef __GNUG__
 #  pragma implementation "pngread.h"
@@ -902,3 +902,4 @@
     return FALSE;
 }
+#endif
 
Index: radiobox.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/radiobox.cpp,v
retrieving revision 1.32
diff -b -u -2 -r1.32 radiobox.cpp
--- radiobox.cpp	2003/09/20 16:24:38	1.32
+++ radiobox.cpp	2004/01/29 19:51:36
@@ -256,5 +256,5 @@
 void wxRadioBox::SetLabel(const wxString& label)
 {
-    return wxControl::SetLabel(label);
+    wxControl::SetLabel(label);
 }
 
@@ -277,5 +277,6 @@
         current=current->NextInCycle();
     }
-    return current->SetLabel(label);
+	
+	current->SetLabel(label);
 }
 
Index: textctrl.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/textctrl.cpp,v
retrieving revision 1.72
diff -b -u -2 -r1.72 textctrl.cpp
--- textctrl.cpp	2003/10/15 19:23:52	1.72
+++ textctrl.cpp	2004/01/29 19:51:36
@@ -21,5 +21,5 @@
   #include <sys/types.h>
   #include <sys/stat.h>
-#else
+#elif !defined(__MRC__)
   #include <stat.h>
 #endif
@@ -46,5 +46,5 @@
 #if defined(__BORLANDC__) && !defined(__WIN32__)
   #include <alloc.h>
-#elif !defined(__MWERKS__) && !defined(__GNUWIN32) && !defined(__DARWIN__)
+#elif !defined(__MWERKS__) && !defined(__GNUWIN32) && !defined(__DARWIN__) && !defined(__MRC__)
   #include <malloc.h>
 #endif
@@ -612,8 +612,11 @@
         TXNTypeAttributes typeAttr[] =
         {
-            {   kTXNQDFontNameAttribute , kTXNQDFontNameAttributeSize , { (void*) fontName } } ,
-            {   kTXNQDFontSizeAttribute , kTXNFontSizeAttributeSize , { (void*) (fontSize << 16) } } ,
-            {   kTXNQDFontStyleAttribute , kTXNQDFontStyleAttributeSize , {  (void*) normal } } ,
+            {   kTXNQDFontNameAttribute , kTXNQDFontNameAttributeSize , NULL /*{ (void*) fontName }*/ } ,
+            {   kTXNQDFontSizeAttribute , kTXNFontSizeAttributeSize , NULL /* { (void*) (fontSize << 16) }*/ } ,
+            {   kTXNQDFontStyleAttribute , kTXNQDFontStyleAttributeSize , NULL /*{  (void*) normal }*/ } ,
         } ;
+		typeAttr[0].data.dataPtr = (void*) fontName;
+		typeAttr[1].data.dataPtr = (void*) (fontSize << 16);
+		typeAttr[2].data.dataPtr = (void*) normal;
 
     err = TXNSetTypeAttributes (varsp->fTXNRec, sizeof( typeAttr ) / sizeof(TXNTypeAttributes) , typeAttr,
Index: timer.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/timer.cpp,v
retrieving revision 1.16
diff -b -u -2 -r1.16 timer.cpp
--- timer.cpp	2003/08/29 00:05:49	1.16
+++ timer.cpp	2004/01/29 19:51:36
@@ -41,5 +41,6 @@
 {
     MacTimerInfo * tm = (MacTimerInfo*)  t ;
-    wxMacAddEvent( tm->m_table , wxProcessTimer, 0 , (void*) tm->m_timer , TRUE ) ;
+/*void wxMacAddEvent( wxMacNotifierTableRef table , wxMacNotificationProcPtr handler , unsigned long event , void* data , short wakeUp ) ;*/
+    wxMacAddEvent( tm->m_table , (wxMacNotificationProcPtr) wxProcessTimer, (unsigned long) 0 , (void*) tm->m_timer , (int) TRUE ) ;
 }
 
Index: utils.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/utils.cpp,v
retrieving revision 1.53
diff -b -u -2 -r1.53 utils.cpp
--- utils.cpp	2003/10/22 17:10:20	1.53
+++ utils.cpp	2004/01/29 19:51:36
@@ -153,5 +153,5 @@
 }
 
-int wxKill(long pid, wxSignal sig , wxKillError *rc )
+int wxKill(long WXUNUSED(pid), wxSignal sig , wxKillError *rc )
 {
     // TODO
@@ -166,5 +166,5 @@
 
 // set the env var name to the given value, return TRUE on success
-WXDLLEXPORT bool wxSetEnv(const wxString& var, const wxChar *value)
+WXDLLEXPORT bool wxSetEnv(const wxString& var, const wxChar *WXUNUSED(value))
 {
     // TODO : under classic there is no environement support, under X yes
Index: window.cpp
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/window.cpp,v
retrieving revision 1.117
diff -b -u -2 -r1.117 window.cpp
--- window.cpp	2003/10/30 23:08:20	1.117
+++ window.cpp	2004/01/29 19:51:36
@@ -1722,5 +1722,6 @@
         {
             wxWindowList hiddenWindows ;
-            for (wxWindowListNode *node = GetChildren().GetFirst(); node; node = node->GetNext())
+			wxWindowListNode *node;
+            for (node = GetChildren().GetFirst(); node; node = node->GetNext())
             {
                 wxControl *child = wxDynamicCast( ( wxWindow*)node->GetData() , wxControl ) ;
@@ -1738,5 +1739,5 @@
             GetEventHandler()->ProcessEvent(event);
  
-            for (wxWindowListNode *node = hiddenWindows.GetFirst(); node; node = node->GetNext())
+            for (node = hiddenWindows.GetFirst(); node; node = node->GetNext())
             {
                 wxControl *child = wxDynamicCast( ( wxWindow*)node->GetData() , wxControl ) ;
Index: wxmac.icns
===================================================================
RCS file: /pack/cvsroots/wxwindows/wxWindows/src/mac/wxmac.icns,v
retrieving revision 1.1
diff -b -u -2 -r1.1 wxmac.icns
Binary files /tmp/cvsTAASEa4Yl and wxmac.icns differ

 	  	 
